.golangci.yml

```diff
@@ -30,104 +26,106 @@ linters:
     - prealloc
     - revive
     - staticcheck
-    - stylecheck
-    - typecheck
     - unconvert
     - unparam
     - unused
     - whitespace
-
-linters-settings:
-  goimports:
-    local-prefixes: github.com/ionos-cloud
-  importas:
-    no-unaliased: true
-    alias:
-      # Kubernetes
-      - pkg: k8s.io/api/(\w+)/(v[\w\d]+)
-        alias: $1$2
-      - pkg: k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1
-        alias: apiextensionsv1
-      - pkg: k8s.io/apimachinery/pkg/apis/meta/v1
-        alias: metav1
-      - pkg: k8s.io/apimachinery/pkg/api/errors
-        alias: apierrors
-      - pkg: k8s.io/apimachinery/pkg/util/errors
-        alias: kerrors
-      # Controller Runtime
-      - pkg: sigs.k8s.io/controller-runtime
-        alias: ctrl
-      - pkg: sigs.k8s.io/cluster-api/api/v1beta1
-        alias: clusterv1
-  loggercheck:
-    require-string-key: true
-    no-printf-like: true
-
+  settings:
+    importas:
+      alias:
+        - pkg: k8s.io/api/(\w+)/(v[\w\d]+)
+          alias: $1$2
+        - pkg: k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1
+          alias: apiextensionsv1
+        - pkg: k8s.io/apimachinery/pkg/apis/meta/v1
+          alias: metav1
+        - pkg: k8s.io/apimachinery/pkg/api/errors
+          alias: apierrors
+        - pkg: k8s.io/apimachinery/pkg/util/errors
+          alias: kerrors
+        - pkg: sigs.k8s.io/controller-runtime
+          alias: ctrl
+        - pkg: sigs.k8s.io/cluster-api/api/v1beta1
+          alias: clusterv1
+      no-unaliased: true
+    loggercheck:
+      require-string-key: true
+      no-printf-like: true
+  exclusions:
+    generated: lax
+    rules:
+      - linters:
+          - gosec
+        text: 'G115: integer overflow conversion'
+      - linters:
+          - gosec
+        text: 'G108: Profiling endpoint is automatically exposed on /debug/pprof'
+      - linters:
+          - revive
+        text: 'exported: exported method .*\.(Reconcile|SetupWithManager|SetupWebhookWithManager) should have comment or be unexported'
+      - linters:
+          - errcheck
+        text: Error return value of .((os\.)?std(out|err)\..*|.*Close|.*Flush|os\.Remove(All)?|.*print(f|ln)?|os\.(Un)?Setenv). is not checked
```

Comment:
you can change this to the `errcheck` settings and exclude functions.

Reply:
This isn't a new or changed check. The file has just been rearranged by the migration tool. I don't want to change any behaviour in this PR, that should go into #403.

Reply:
see [line 89](https://github.com/ionos-cloud/cluster-api-provider-proxmox/pull/468/files/38e42bd42d3d22241d2e753e07a89a8aa3fa7769#diff-6179837f7df53a6f05c522b6b7bb566d484d5465d9894fb04910dd08bb40dcc9L89)